### PR TITLE
Check all user id group pairs inside a security group rule

### PR DESCRIPTION
**Pull request by hanwen-pcluste**
### Description of changes
* Describe *what* you're changing and *why* you're doing these changes.
The assumption of "UserIdGroupPairs is always of length 1" is wrong. This led to false validator failures.
* Link to impacted open issues.
https://github.com/aws/aws-parallelcluster/issues/3590

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ CHANGELOG
 - Fix exporting of cluster logs when there is no prefix specified, previously exported to a `None` prefix.
 - Fix rollback not being performed in case of cluster update failure.  
 - Fix RootVolume schema for the HeadNode.
+- Fix EfaSecurityGroupValidator. Previously, it may produce false failures when custom security groups were provided and EFA was enabled.
```
Comment (enrico-usai) [inline CHANGELOG.md:28]: What about something like: `Fix EfaSecurityGroupValidator to correctly validate custom Security Groups with more than 1 UserIdGroupPairs when EFA is enabled.`

File: cli/tests/pcluster/validators/test_cluster_validators.py [hunk 2]
```diff
@@ -242,6 +242,31 @@ def test_efa_placement_group_validator(
             [{"IpProtocol": "-1", "UserIdGroupPairs": [{"UserId": "123456789012", "GroupId": "sg-12345678"}]}],
             None,
         ),
+        # right sg. Test when UserIdGroupPairs contains more entries
```
Comment (enrico-usai) [inline cli/tests/pcluster/validators/test_cluster_validators.py:245]: Do we have an integration test with custom Security Group? If so I'd extend it to have more than 1 group pair.

File: CHANGELOG.md (hunk 1 above)
Comment (hanwen-pcluste) [reply CHANGELOG.md:28]: I tried to use more user-friendly wordings in the change log, and more technical wordings in the commit description. But I am ok to change the message here

File: cli/tests/pcluster/validators/test_cluster_validators.py (hunk 2 above)
Comment (hanwen-pcluste) [reply cli/tests/pcluster/validators/test_cluster_validators.py:245]: We do. But adding another group pair does not guarantee we could discover the issue from the integration tests. If the boto3 call returns the group pairs with the correct order, the test would not discover the issue. Moreover, it feels like a unit test to validate our validators is enough

**Review by enrico-usai — APPROVED** [review]

**Pull request merged**